src/Microsoft.AspNetCore.StaticFiles/CacheProfile.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.StaticFiles
+{
+    /// <summary>
+    /// Defines settings used for caching responses for the <see cref="StaticFileMiddleware"/>.
+    /// </summary>
+    public class CacheProfile
+    {
+        /// <summary>
+        /// Gets or sets the duration in seconds for which the response is cached.
+        /// Sets the "max-age" in the "Cache-control" header in the 
+        /// <see cref="Microsoft.AspNetCore.Http.HttpContext.Response" />.
+        /// Defaults to 10 minutes
+        /// </summary>
+        public int Duration { get; set; } = 600;
+
+        /// <summary>
+        /// Gets or sets the location where the data from a particular URL must be cached.
+        /// Sets the "Cache-control" header in the 
+        /// <see cref="Microsoft.AspNetCore.Http.HttpContext.Response" />. 
+        /// Defaults to  <see cref="ResponseCacheLocation.Any" />
+        /// </summary>
+        public ResponseCacheLocation Location { get; set; } = ResponseCacheLocation.Any;
```

Comment:
Won't that be confusing for people? By default `CacheProfile` will be null, so it is already opt in. If we leave it as any, then again it's a similar interface to the `ResponseCacheAttribute`, so that for example:

`options.CacheProfile = new CacheProfile { Duration = 60 }; ` 

would set up `public,max-age=60`. If we change to `None` then you have to do 

`options.CacheProfile = new CacheProfile { Duration = 60, Location = ResponseCacheLocation.Any}; `

which seems less obvious to me?